executor.py

```diff
@@ -18,17 +18,21 @@ class SimpleIndexer(Executor):
     def __init__(
         self,
         match_args: Optional[Dict] = None,
+        key_length: int = 64,
+        override_storage_path: Optional[str] = None,
         **kwargs,
     ):
         """
         Initializer function for the simple indexer
         :param match_args: the arguments to `DocumentArray`'s match function
+        :param override_storage_path: the `path` argument to `DocumentArrayMemmap`'s constructor
+        :param key_length: the `key_length` keyword argument to `DocumentArrayMemmap`'s constructor
         """
         super().__init__(**kwargs)
 
         self._match_args = match_args or {}
         self._storage = DocumentArrayMemmap(
-            self.workspace, key_length=kwargs.get('key_length', 64)
+            override_storage_path or self.workspace, key_length = key_length
```

Comment:
On second thought, if we set the` workspace` path when initializing `SimpleIndexer` as follows:
```python
Flow.add(uses=`jinahub://SimpleIndexer', uses_metas={'workspace': workspace})
```
then we don't need `override_storage_path`?

Reply:
Yes, technically.  
But using workspace as storage root is an implementation detail to users. I did not know how to override this path before looking into the source code. 
And as a newbie, I also did not know that workspace could be overridden using `uses_metas`...
So I think having a `uses_with` parameter for path overriding is a reasonable approach and is more intuitive to users (who don't know all the internals).

Reply:
reject, this should be a documentation job